main.cc

```diff
@@ -357,12 +357,39 @@ int main(int ac, char** av) {
     // If --version is requested, print it out and exit immediately to avoid
     // Seastar-specific warnings that may occur when running the app
     bpo::variables_map vm;
-    bpo::store(bpo::command_line_parser(ac, av).options(app.get_options_description()).allow_unregistered().run(), vm);
+    auto parsed_opts = bpo::command_line_parser(ac, av).options(app.get_options_description()).allow_unregistered().run();
+    bpo::store(parsed_opts, vm);
     if (vm["version"].as<bool>()) {
         fmt::print("{}\n", scylla_version());
         return 0;
     }
 
+    if (ac) {
+        fmt::print("Scylla executed using: \"{}", av[0]);
+        for (int i = 1; i < ac; ++i) {
+            fmt::print(" {}", av[i]);
+        }
+        fmt::print("\"\n");
+    }
+
+    fmt::print("Parsed command line options:\n");
+    for (auto const& opt: parsed_opts.options) {
+        if (opt.value.empty()) {
+            fmt::print("{}", opt.string_key);
+        } else {
+            if (opt.string_key.empty()) {
+                fmt::print("(positional)");
+            } else {
+                fmt::print("{}:", opt.string_key);
+            }
+
+            for (auto const& val: opt.value) {
+                fmt::print(" {}", val);
+            }
+        }
+        fmt::print("\n");
+    }
```

Comment:
That's very verbose, better to append all the parameters to the initial log message ("Scylla version {} starting"). You can use the argc/argv, no need to mess with the parsed options.

Reply:
I wanted to log how exactly the options were parsed --- if I understand correctly, that was the point of issue #4203. Indeed, issue #4203 references issue #4141, in which a user provides the command used to run Scylla, but that information doesn't entirely help with debugging the problem; the issue arises from incorrect parsing of the command's arguments. So I think it's useful to know this.
However I agree that it's verbose. I will change it so that the parsed arguments are printed in a readable format in a single line.

Reply:
Done.